libs/klv/src/test/java/org/codice/alliance/libs/klv/AlphanumericDistinctKlvProcessorTest.java

```diff
@@ -0,0 +1,71 @@
+/**
+ * Copyright (c) Codice Foundation
+ *
+ * <p>This is free software: you can redistribute it and/or modify it under the terms of the GNU
+ * Lesser General Public License as published by the Free Software Foundation, either version 3 of
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.alliance.libs.klv;
+
+import static org.hamcrest.CoreMatchers.equalTo;
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.CoreMatchers.not;
+import static org.hamcrest.CoreMatchers.nullValue;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import com.google.common.collect.ImmutableList;
+import ddf.catalog.data.Attribute;
+import ddf.catalog.data.Metacard;
+import ddf.catalog.data.impl.MetacardImpl;
+import java.io.Serializable;
+import java.util.List;
+import org.junit.Test;
+
+public class AlphanumericDistinctKlvProcessorTest {
```

Comment:
How about a test for when the KLV data contains random binary data instead of proper alphanumeric data?

Reply:
What exactly do you mean? Do you mean something different than what I'm testing in the `doProcessWithNonalphanumericValues` test? If so, could you give me an example?

Reply:
Well, it looks like the code must deal with Serializable, so it may not be possible to pass in just random binary data. I was thinking of the case where an unknown character encoding was used. But this is probably the wrong place to handle it anyways. nm